dashpay/src/main/kotlin/org/dashj/platform/dashpay/BlockchainIdentity.kt

```diff
@@ -642,6 +655,7 @@ class BlockchainIdentity {
 
     fun createPreorderDocuments(unregisteredUsernames: List<String>): List<Document> {
         val usernamePreorderDocuments = ArrayList<Document>()
+        checkIdentity()
         for (saltedDomainHash in saltedDomainHashesForUsernames(unregisteredUsernames).values) {
             val document = platform.names.createPreorderDocument(Sha256Hash.wrap(saltedDomainHash), identity!!)
```

Comment:
If `initializeIdentity` fails and gets into `catch (x: Exception)` (see above) and `identity` isn't assigned, we're going to get NRE here, which would be less informative.

Reply:
The original state was to get an `NPE` if `identity == null`.  I thought these changes were helping to fix an exception and crash in `dash-wallet/dashpay`, but now I realize that this doesn't actually fix the crash that I was trying to fix.

`identity` would be null if there was a network error when attempting to obtain the identity structure from Platform.  

With DashPay being used by many users, we haven't seen a crash from `identity == null`.  It would be better to perhaps throw an exception with `checkIdentity()` that is more useful than NPE.